### PR TITLE
lightning: refine progress for compress files import

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/38514

Problem Summary:

### What is changed and how it works?
Currently lightning's progress is not accurate for compress files import. This PR uses reader's posistion to calculate progress for compressed files.

### Check List

Tests 

- [x] Manual test (add detailed scripts or steps below)
  check whether lightning progress is correct when importing compressed files.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/mydump/region.go [hunk 1]
```diff
@@ -302,7 +302,7 @@ func MakeSourceFileRegion(
 	// set fileSize to INF to make sure compressed files can be read until EOF. Because we can't get the exact size of the compressed files.
 	// TODO: update progress bar calculation for compressed files.
 	if fi.FileMeta.Compression != CompressionNone {
-		rowIDMax = fileSize * 100 / divisor // FIXME: this is not accurate. Need more tests and fix solution.
+		rowIDMax = fileSize * 500 / divisor // FIXME: this is not accurate. Need more tests and fix solution.
```
Comment (buchuitoudegou) [inline br/pkg/lightning/mydump/region.go:305]: Does `500` represent the compress ratio? Maybe leave a comment here to illustrate the number.

**Review by buchuitoudegou — APPROVED** [review]

File: br/pkg/lightning/restore/restore.go [hunk 2]
```diff
@@ -2618,14 +2632,22 @@ func (cr *chunkRestore) encodeLoop(
...
+		var newOffset, rowID, realOffset int64
 		var kvSize uint64
+		var realOffsetErr error
 	outLoop:
 		for !canDeliver {
 			readDurStart := time.Now()
 			err = cr.parser.ReadRow()
 			columnNames := cr.parser.Columns()
 			newOffset, rowID = cr.parser.Pos()
+			if cr.chunk.FileMeta.Compression != mydump.CompressionNone {
```
Comment (dsdashun) [inline br/pkg/lightning/restore/restore.go:2644]: If not compressed, the `realOffset` is not assigned and 0 is used.  Is this the expected behavior ?

File: br/pkg/lightning/restore/restore.go [hunk 3]
```diff
@@ -2299,6 +2299,8 @@ type deliveredKVs struct {
 	columns []string
 	offset  int64
 	rowID   int64
+
+	realOffset int64
```
Comment (dsdashun) [inline br/pkg/lightning/restore/restore.go:2303]: Maybe add some comment, indicating that `realOffset` is only used in compressed file scenarios.

**Review by dsdashun — COMMENTED** [review]
rest LGTM

File: br/pkg/lightning/restore/restore.go (hunk 3 above)
Comment (lichunzhu) [reply br/pkg/lightning/restore/restore.go]: addressed in https://github.com/pingcap/tidb/pull/39219/commits/32071eda886a93da125f95529c2a4cf334864ade

**Review by dsdashun — APPROVED** [review]
LGTM

**Pull request merged**